src/OpenTracing.MockTracer/MockTracer.cs

```diff
@@ -0,0 +1,225 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using OpenTracing.Propagation;
+
+namespace OpenTracing.MockTracer
+{
+    /// <summary>
+    /// In-memory <see cref="ITracer"/> implementation designed to make it
+    /// easy to test assertions against the OpenMetrics API and its use
+    /// inside frameworks.
+    /// </summary>
+    public class MockTracer : ITracer
+    {
+        private ConcurrentQueue<MockSpan> _finishedSpans = new ConcurrentQueue<MockSpan>();
```

Comment:
[nit] It is possible to make this `readonly` if using locks like java uses `synchronized` for access to this structure, but probably not worth the effort (e.g. who really cares if this private field is `readonly`, the cases where it matters are pretty small).

Reply:
I removed the Concurrent* stuff and went back to regular Lists and locking to be closer to Java. The PR was missing some locks anyway.